### PR TITLE
Add ssh key pair bait to sandbox

### PR DESCRIPTION
**Pull request by elainechien**
This is PR is part of https://github.com/ossf/package-analysis/issues/586

### CODE REVIEW

File: internal/worker/rundynamic.go [hunk 1]
```diff
@@ -120,6 +124,37 @@ func retrieveExecutionLog(ctx context.Context, sb sandbox.Sandbox) (string, erro
 	return string(processedLog), nil
 }
 
+func addSSHKeysToSandbox(sb sandbox.Sandbox, ctx context.Context) error {
```
Comment (maxfisher-g) [inline internal/worker/rundynamic.go:127]: could you add a comment describing the function? (should start with the function name as per go doc standards)

File: internal/worker/rundynamic.go [hunk 2]
```diff
@@ -120,6 +124,37 @@ func retrieveExecutionLog(ctx context.Context, sb sandbox.Sandbox) (string, erro
...
+	generatedPrivateKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	generatedPublicKey := generatedPrivateKey.PublicKey
+	if err != nil {
+		return err
+	}
+
+	os.Mkdir("temp_ssh_dir", 0750)
```
Comment (maxfisher-g) [inline internal/worker/rundynamic.go:134]: prefer `tempdir, err := os.MkdirTemp(".", "temp_ssh_dir")` so the dirname is in a variable.

Then you can use `filepath.Join` in expressions below to create the files in the temp directory

File: internal/worker/rundynamic.go (hunk 1 above)
Comment (calebbrown) [inline internal/worker/rundynamic.go:127]: If context is present in an argument list, it is always first (see https://google.github.io/styleguide/go/decisions#contexts)

File: internal/worker/rundynamic.go (hunk 2 above)
Comment (calebbrown) [reply internal/worker/rundynamic.go:134]: Also defer a removal of the `tempdir`

File: internal/worker/rundynamic.go (hunk 1 above)
Comment (elainechien) [reply internal/worker/rundynamic.go]: Done.

Comment (elainechien) [reply internal/worker/rundynamic.go]: Done.

File: internal/worker/rundynamic.go (hunk 2 above)
Comment (elainechien) [reply internal/worker/rundynamic.go]: Done.

File: internal/worker/rundynamic.go [hunk 3]
```diff
@@ -163,6 +205,10 @@ func RunDynamicAnalysis(ctx context.Context, pkg *pkgmanager.Pkg, sbOpts []sandb
 		return DynamicAnalysisResult{}, err
 	}
 
+	if err := addSSHKeysToSandbox(ctx, sb); err != nil {
+		return DynamicAnalysisResult{}, err
```
Comment (maxfisher-g) [inline internal/worker/rundynamic.go:209]: I'm not sure if we should fail the whole of dynamic analysis if adding SSH keys fails. Perhaps we should just log the error and proceed without SSH keys. Thoughts @calebbrown?

**Review by maxfisher-g — COMMENTED** [review]
Nice work Elaine! Just have one design point to figure out, see below

File: internal/worker/rundynamic.go (hunk 3 above)
Comment (elainechien) [reply internal/worker/rundynamic.go:209]: Ah yeah just logging and proceeding without SSH keys makes sense to me.

Comment (elainechien) [reply internal/worker/rundynamic.go]: Done.

Comment (maxfisher-g) [reply internal/worker/rundynamic.go]: Thanks!

**Review by maxfisher-g — APPROVED** [review]

**Pull request merged**